src/main/java/com/booleanuk/Poker.java

```diff
@@ -1,25 +1,117 @@
+            String[] _pair = findPair(hand);
+            if (_pair.length > 0)
+                _potentialHands.add(hand);
+        }
+
+        ArrayList<String[]> _threeOfAKind = new ArrayList<>();
+        ArrayList<String[]> _pairs = new ArrayList<>();
+
+        for (String[] handPairs : _potentialHands) {
+            String _cardToCompare = handPairs[0];
+            boolean _isTOAK = handPairs.length > 2;
+
+            for (String card : handPairs) {
+                if (!_cardToCompare.equals(card)) {
+                    _isTOAK = false;
+                    break;
+                }
+            }
+
+            if (_isTOAK) {
+                System.out.println(Arrays.toString(handPairs));
+                _threeOfAKind.add(handPairs);
+            }
+            else
+                _pairs.add(handPairs);
+        }
+
+        return getWinningHandOnPairs(_threeOfAKind.isEmpty() ? _pairs : _threeOfAKind);
+    }
+
+    private String[] getWinningHandOnPairs(ArrayList<String[]> pairs) {
+        int _highestScore = 0;
+        String[] _winningHand = new String[0];
+
+        for (String[] hand : pairs) {
+            int _score = getHandScore(hand);
+            if (_score > _highestScore) {
+                _highestScore = _score;
+                _winningHand = hand;
+            }
+        }
+
+        return _winningHand;
+    }
+
+    private int getHandScore(String[] hand) {
+        int _outScore = 0;
+        for (String card : hand) {
+            for (int i = 0; i < cardPriorityOrder.length; i++)
+                if (cardPriorityOrder[i].equals(card))
+                    _outScore += i;
+        }
+        return _outScore;
+    }
+
     // Core
     public String[] winningPair(String[] firstHand, String[] secondHand) {
         // Implement the winningPair logic here and return the array containing the winning pair to make the tests pass.
         // You can replace the following return value with something appropriate
-        return new String[]{"Replace me", "with something else"};
+
+        return highestPair(new String[][]{ firstHand, secondHand });
     }
 
     // Extension 1
     public String[] winningPairFromArray(String[][] hands) {
         // Implement the winningPairFromArray logic here and return the array containing the winning pair to make the tests pass.
         // You can replace the following return value with something appropriate
-        return new String[]{"Replace me", "with something else"};
+
+        return highestPair(hands);
     }
 
     // Extension 2
     public String[] winningThreeCardHand(String[][] hands) {
         // Implement the winningThreeCardHand logic here and return the array containing the winning hand to make the tests pass.
         // You can replace the following return value with something appropriate
-        return new String[]{"Replace me", "with something else"};
+        return highestPair(hands);
     }
 
     // Extension 3
```

Comment:
Good approach identifying pairs in the `winningPair` method. Your implementation of `highestPair` for both Extension 1 and Extension 2 shows a solid understanding and a good attempt at handling multiple scenarios. Nice work.